### DISCUSSION TITLE
Dynamically create MenuFlyouts

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#15273] [Q&A] [answered]
I want to create a `MenuFlyout` for a button dynamically based on a list of custom types. I found an example for a regular `Menu` (https://docs.avaloniaui.net/docs/0.10.x/controls/menu#dynamically-creating-menus). But unfortunately it seems not to work for a `MenuFlyout`.

Is there something equivalent for a `MenuFlyout`?

```Xaml
<Menu Items="{Binding MenuItems}">
    <Menu.Styles>
        <Style Selector="MenuItem">
            <Setter Property="Header" Value="{Binding Header}"/>
            <Setter Property="Items" Value="{Binding Items}"/>
            <Setter Property="Command" Value="{Binding Command}"/>
            <Setter Property="CommandParameter" Value="{Binding CommandParameter}"/>
        </Style>
    </Menu.Styles>
</Menu>
```

I a looking for a solution for Avalonia 11.x

**Comment by timunie** [comment]
`<MenuFlyout ItemsSource="{Binding MyItems}"></MenuFlyout>` works the same, no?

**Reply by DevAM-Tools** [reply]
> `<MenuFlyout ItemsSource="{Binding MyItems}"></MenuFlyout>` works the same, no?

That's working fine but I get a style applied to the items (Header or Command etc.) so that I only see the class name for each item.

**Reply by timunie** [reply]
you can add the styles to it or override ToString in your ItemModel

**Reply by DevAM-Tools** [reply]
Can you give me a hint how to add a style? I was not able to figure it out yet.

**Comment by workgroupengineering** [comment, marked as answer]
> I want to create a `MenuFlyout` for a button dynamically based on a list of custom types. I found an example for a regular `Menu` (https://docs.avaloniaui.net/docs/0.10.x/controls/menu#dynamically-creating-menus). But unfortunately it seems not to work for a `MenuFlyout`.

This document has been outbanded , from Avalonia 11.x.x you should do this

```xaml
      <MenuFlyout ItemsSource="{Binding MenuItems}">
        <MenuFlyout.ItemContainerTheme>
          <ControlTheme TargetType="MenuItem" BasedOn="{StaticResource {x:Type MenuItem}}"
                        x:DataType="l:MenuItemViewModel">
            <Setter Property="Header" Value="{Binding Header}"/>
            <Setter Property="ItemsSource" Value="{Binding Items}"/>
            <Setter Property="Command" Value="{Binding Command}"/>
            <Setter Property="CommandParameter" Value="{Binding CommandParameter}"/>
          </ControlTheme>
        </MenuFlyout.ItemContainerTheme>
      </MenuFlyout>
```

I attach a complete [example](https://github.com/AvaloniaUI/Avalonia/files/14945024/AMenuFlyout.zip).

**Reply by DevAM-Tools** [reply]
Awesome! This was the missing piece of the puzzle. Thank you very much!

**Reply by DevAM-Tools** [reply]
I put this to a [PR](https://github.com/AvaloniaUI/avalonia-docs/pull/451) for the documentation.
